maestro-cli/src/main/java/maestro/cli/App.kt

```diff
@@ -63,10 +64,13 @@ class App {
     @Option(names = ["--host"], hidden = true)
     var host: String? = null
 
+    @Option(names = ["--driver-host-port"], hidden = true)
+    var driverHostPort: Int = Random.nextInt(9001, 9128)
```

Comment:
no worries, this is a hidden parameter.

I wrote a more detailed explanation of how this makes up one of the three core parts of the patch for your question `Why the sudden change of port` . Since it's a hidden parameter, users don't have to worry about it, unless they want to specify the port themselves, as an advanced user. 

It is only an initial runtime configuration for the driver, which gets randomised so that it allows multiple maestros and multiple android drivers to run simultaneously in parallel

Reply:
It's perhaps a fringe use case, but you _would_ need to vary the driver host port if you were wishing to use Maestro with a custom, externalised sharding solution using multiple maestro instances + emulators in the same OS, since this is the only port which would clash between the maestro's.

Reply:
🤔 with the most recent changes I'm actually randomising a not-already-used port within 9000-9128 range for each shard, as I edited my post

So there is no need to specify this any more, logic will override it in newSession anyways, I'll refactor this to be a private variable, doesn't even need to be a CLI parameter. Easy peasy

Reply:
with the most recent changes, when sharding is enabled this port is now randomised between the 9000-9128 range. 

With no sharding I'm going to keep the default global value of 7001 for old users rev compatibility, but if sharding is enabled it will allocate from between the 9000-9128 range as before.

Reply:
right, I realised this parameter variable itself is unnecessary as newSession either takes --port, or default 7001 for reverse compatibility, or if sharding is on chooses an unused port from range 7001-7128 . I removed this seperate CLI variable, the logic is already contained inside newSession or App's --port

Reply:
```
val driverHostPort = if (!sharded) parent?.port ?: 7001 else
                        (7001..7128).shuffled().find { port ->
                            usedPorts.putIfAbsent(port, true) == null
                        } ?: error("No available ports found")
```
✅